### PR TITLE
fix device update

### LINKED ISSUE #3500: Renaming a device is inconsistent
### Description

I noticed that renaming a device using `PUT /api/devices/UID` isn't consistent.

Sometimes, the device will not be renamed. Sometimes a 409 will be thrown when you try to rename a device with a name already used, but not always.

To reproduce, just have a shellhub instance with one instance on it and spam a rename with a new name on it. The name will not change directly, and once it finally does, then it will not always throw a 409

### Edition

Community

### Version

0.14.1

### PR DESCRIPTION
**Pull request by heiytor**
fixes #3500

### CODE REVIEW

File: api/routes/device.go [hunk 1]
```diff
@@ -322,9 +322,10 @@ func (h *Handler) UpdateDevice(c gateway.Context) error {
 		tenant = c.Tenant().ID
 	}
 
-	if err := guard.EvaluatePermission(c.Role(), guard.Actions.Device.Update, func() error {
+	err := guard.EvaluatePermission(c.Role(), guard.Actions.Device.Update, func() error {
 		return h.service.UpdateDevice(c.Ctx(), tenant, models.UID(req.UID), req.Name, req.PublicURL)
-	}); err != nil {
+	})
+	if err != nil {
```
Comment (henrybarreto) [inline api/routes/device.go:328]: Why do you have preferred changed how the if was structured?

Comment (heiytor) [reply api/routes/device.go:328]: I strongly believe that the syntax of `if err := ...; err != nil {}` becomes highly illegible when the `[..]; err != nil {}` statement is placed on another line or when the function call is too long. In my opinion, the eyes get lost when trying to understand the flow of the code.

Comment (heiytor) [reply api/routes/device.go:328]: This has been my experience, and it is why I consider it a good idea to change this.

Comment (otavio) [reply api/routes/device.go:328]: I believe binding it to the `if` statement also reduces the variable's scope. I prefer the original code (even if it's uglier).

Comment (henrybarreto) [reply api/routes/device.go:328]: > I believe binding it to the `if` statement also reduces the variable's scope. I prefer the original code (even if it's uglier).

I agree. This change could fit perfectly if we needed the error in the following, but we don't.

File: api/store/mongo/device.go [hunk 2]
```diff
@@ -536,31 +536,26 @@ func (s *Store) DeviceChooser(ctx context.Context, tenantID string, chosen []str
...
-// excluding devices with UIDs present in the "notIn" list.
 func (s *Store) DeviceUpdate(ctx context.Context, tenant string, uid models.UID, name *string, publicURL *bool) error {
-	session, err := s.db.Client().StartSession()
-	if err != nil {
-		return err
+	// Not deleting the device from the cache may cause issues when trying to retrieve the device after the update.
+	// TODO: Maybe we can standardize the key creation?
+	if err := s.cache.Delete(ctx, strings.Join([]string{"device", string(uid)}, "/")); err != nil {
+		logrus.Error(err)
 	}
```
Comment (henrybarreto) [inline api/store/mongo/device.go:544]: Delete the cache is an excellent idea, but I think that we should do it after a successful update only

File: api/routes/device.go (hunk 1 above)
Comment (heiytor) [reply api/routes/device.go]: I understand your concerns; I'm rolling back this commit.

**Review by henrybarreto — APPROVED** [review]

**Pull request merged**